stacks/api/function/webhook/test/invoker.spec.ts

```diff
@@ -0,0 +1,131 @@
+import {WebhookService} from "@spica-server/function/webhook";
+import {WebhookInvoker} from "@spica-server/function/webhook/src/invoker";
+import * as __fetch__ from "node-fetch";
+
+describe("Webhook Invoker", () => {
+  let invoker: WebhookInvoker;
+  let module: TestingModule;
+  let service: WebhookService;
+  let db: DatabaseService;
+
+  let subscribeSpy: jasmine.Spy<typeof invoker["subscribe"]>;
+  let unsubscribeSpy: jasmine.Spy<typeof invoker["unsubscribe"]>;
+  let fetchSpy: jasmine.Spy<typeof __fetch__.default>;
+
+  beforeEach(async () => {
+    module = await Test.createTestingModule({
+      imports: [DatabaseTestingModule.replicaSet()],
+      providers: [WebhookInvoker, WebhookService]
+    }).compile();
+
+    service = module.get(WebhookService);
+    db = module.get(DatabaseService);
+
+    invoker = module.get(WebhookInvoker);
+    await stream.wait();
+
+    subscribeSpy = spyOn(invoker, "subscribe" as never).and.callThrough();
+    unsubscribeSpy = spyOn(invoker, "unsubscribe" as never).and.callThrough();
+    fetchSpy = spyOn(__fetch__, "default").and.returnValue(Promise.resolve(undefined));
+  });
+
+  afterEach(async () => await module.close());
+
+  it("should subscribe and open a change stream against the collection", async () => {
+    const {_id, ...hook} = await service.insertOne({
+      url: "http://spica.internal",
+      trigger: {
+        name: "database",
+        active: true,
+        options: {
+          collection: "stream_coll",
+          type: "INSERT"
+        }
+      }
+    });
+    await stream.change.wait();
+    const subsequentStream = await stream.wait();
+    expect(subscribeSpy).toHaveBeenCalledTimes(1);
+    expect(subscribeSpy).toHaveBeenCalledWith(_id.toHexString(), hook);
+    expect(subsequentStream).toEqual([
+      "stream_coll",
+      [
+        {
+          $match: {operationType: "insert"}
+        }
+      ],
+      {fullDocument: "updateLookup"}
+    ]);
+  });
+
+  it("should unsubscribe after the webhook has deleted", async () => {
+    const hook = await service.insertOne({
+      url: "http://spica.internal",
+      trigger: {
+        name: "database",
+        active: true,
+        options: {
+          collection: "stream_coll",
+          type: "INSERT"
+        }
+      }
+    });
+    await stream.change.wait();
+    await service.deleteOne({_id: hook._id});
+    await stream.change.wait();
+    expect(unsubscribeSpy).toHaveBeenCalledTimes(1);
+    expect(unsubscribeSpy).toHaveBeenCalledWith(hook._id.toHexString());
+  });
+
+  it("should unsubscribe after the webhook has disabled", async () => {
+    const hook = await service.insertOne({
+      url: "http://spica.internal",
+      trigger: {
+        name: "database",
+        active: true,
+        options: {
+          collection: "stream_coll",
+          type: "INSERT"
+        }
+      }
+    });
+    await stream.change.wait();
+    await service.updateOne({_id: hook._id}, {$unset: {"trigger.active": ""}});
```

Comment:
Also, we should change this line too, if we will change the switch cases that I wrote in my previous commit.